node/flatpak_node_generator/providers/npm.py

```diff
@@ -142,6 +143,8 @@ def __init__(
         self.git_sources: DefaultDict[
             Path, Dict[Path, GitSource]
         ] = collections.defaultdict(lambda: {})
+        # FIXME better pass the same provider object we created in main
```

Comment:
this hurts :cry: but it's tricky to change, I'm playing around a bit with it rn but we can definitely just clean it all up later.

Reply:
It does. TBH while implementing this, I started to doubt if adding a separate factory class for rcfiles was a good idea at all. But ofc this is out of scope of this PR.